### PR TITLE
[GO-84] New stats

### PR DESCRIPTION
**Pull request by im-kulikov**
- doc-comments
- typos
- errors to vars
- new stats (`alloc_heap`, `alloc_heap_sys`, `mallocs`, `sys`, `frees`, `threads`)

---

Pls, bump new tag 2.1 or 3.0 as you wish

### CODE REVIEW

**Review by pcherednichenko — APPROVED** [review]

**Review by jenchik — APPROVED** [review]

File: metrics.go [hunk 1]
```diff
@@ -132,11 +152,11 @@ func New(url string, application, hostname string, options ...nats.Option) (*con
 
 	// Getting current environment
 	if application == "" {
-		return nil, errors.New("Application name not set")
+		return nil, ErrAppName
```
Comment (m1ome) [inline metrics.go]: `ErrAppNameNotSet`

File: metrics_test.go [hunk 2]
```diff
@@ -39,6 +39,10 @@ func TestMain(m *testing.M) {
 		natsURL = nats.DefaultURL
 	}
 
+	SetErrorHandler(func(err error) {
```
Comment (m1ome) [inline metrics_test.go:42]: Wut? O.O

**Review by m1ome — CHANGES_REQUESTED** [review]

File: metrics_test.go (hunk 2 above)
Comment (im-kulikov) [reply metrics_test.go:42]: just for tests

Comment (im-kulikov) [reply metrics_test.go:42]: if we have some errors on tests we do panic, why not?

**Review by m1ome — APPROVED** [review]

**Pull request merged**